src/redux-toolkit/slices/homePageSlice.ts

```diff
@@ -1,44 +1,131 @@
 import { PayloadAction, createSlice } from "@reduxjs/toolkit";
-import Data, { Chef, Dish, Restaurant } from "../../data/types/data";
+import { Chef, Dish, Restaurant } from "../../data/types/data";
+import {
+  getPopularRestaurants,
+  getChefOfTheWeek,
+  getSignatureDish,
+  getChefs,
+} from "../thunks/homePageThunk";
+import { rmdirSync } from "fs";
 
-interface HomePageState{
-    selectedDish: Dish | null,
-    isOpen: boolean,
-    dataRestaurants: Restaurant[],
-    dataDishes: Dish[],
-    dataChefOfTheWeek: Chef
+interface HomePageState {
+  selectedDish: Dish | null;
+  isOpen: boolean;
+  dataRestaurants: Restaurant[];
+  dataDishes: Dish[];
+  dataChefOfTheWeek: Chef;
+  dataChefs: Chef[];
+}
 
+const initialState: HomePageState = {
+  selectedDish: null,
+  isOpen: false,
+  dataRestaurants: [],
+  dataDishes: [],
+  dataChefOfTheWeek: { img: "", name: "", about: "", chefRestaurants: [], _id: "" },
+  dataChefs: [],
 };
 
-const initialState: HomePageState={
-    selectedDish: null,
-    isOpen: false,
-    dataRestaurants: [],
-    dataDishes: [],
-    dataChefOfTheWeek: {img:"", name:"", about:"", chefRestaurants:[]},
-};
+const homePageSlice = createSlice({
+  name: "homePage",
+  initialState: initialState,
+  reducers: {
+    setSelectedDish(state, action: PayloadAction<Dish | null>) {
+      state.selectedDish = action.payload;
+    },
+    setIsOpen(state, action: PayloadAction<boolean>) {
+      state.isOpen = action.payload;
+    },
+    setDataRestaurants(state, action: PayloadAction<Restaurant[]>) {
+      state.dataRestaurants = action.payload;
+    },
+    setDataDishes(state, action: PayloadAction<Dish[]>) {
+      state.dataDishes = action.payload;
+    },
+    setDataChefOfTheWeek(state, action: PayloadAction<Chef>) {
+      state.dataChefOfTheWeek = action.payload;
+    },
+  },
+
+  extraReducers(builder) {
+    builder
+      .addCase(getPopularRestaurants.fulfilled, (state, action: PayloadAction<any>) => {
+        const restaurants = action.payload.map((res: any) => {
+          return {
+            img: res.img,
+            name: res.name,
+            chef: state.dataChefs.find((chef: Chef) => chef.chefRestaurants.includes(res._id)),
```

Comment:
The chef should come as a populate and not wait for the call of the chefs

Reply:
fixed